worker/caasfirewallersidecar/applicationworker.go

```diff
@@ -165,6 +156,20 @@ func (w *applicationWorker) loop() (err error) {
 			if !ok {
 				return errors.New("application watcher closed")
 			}
+
+			// If charm is (now) a v1 charm, exit the worker.
+			format, err := w.charmFormat()
+			if errors.IsNotFound(err) {
+				w.logger.Debugf("application %q removed", w.appName)
```

Comment:
`application bla removed` sounds like the application was removed by this worker. `application bla no longer exists` something like that.

Reply:
Good call, done.